### ISSUE TITLE
Add arithmetic fixpoint function

### ISSUE THREAD

**Issue by Technologicat** [Technologicat/unpythonic#14]
Dumping old notes. This needs stuff from `unpythonic.it`. Maybe put it there, or just for disambiguation purposes, into `unpythonic.fix`?

Note we should also switch the order of arguments of `within`; the way it's currently implemented, it's not curry-friendly, while this assumes a curry-friendly implementation. The price is no default tol, but this just simplifies the implementation, so maybe it's a good idea. Also `within(tol, iterable)` is better English than `within(iterable, tol)`.

```python
def fixpoint(f, x0, tol=0):
    """Compute the (arithmetic) fixed point of f, starting from the initial guess x0.

    (Not to be confused with the logical fixed point with respect to the
    definedness ordering.)

    The fixed point must be attractive for this to work. See the Banach
    fixed point theorem.
    https://en.wikipedia.org/wiki/Banach_fixed-point_theorem

    If the fixed point is attractive, and the values are represented in
    floating point (hence finite precision), the computation should
    eventually converge down to the last bit (barring roundoff or
    catastrophic cancellation in the final few steps). Hence the default tol
    of zero.

    CAUTION: an arbitrary function from ℝ to ℝ **does not** necessarily
    have a fixed point. Limit cycles and chaotic behavior of `f` will cause
    non-termination. Keep in mind the classic example:
    https://en.wikipedia.org/wiki/Logistic_map

    Examples::
        import math
        from unpythonic import fixpoint
        c = fixpoint(math.cos, x0=1)

        # Actually "Newton's" algorithm for the square root was already known to the
        # ancient Babylonians, ca. 2000 BCE. (Carl Boyer: History of mathematics)
        def sqrt_newton(n):
            def sqrt_iter(x):  # has an attractive fixed point at sqrt(n)
                return (x + n / x) / 2
            return fixpoint(sqrt_iter, x0=n / 2)
        s = sqrt_newton(2)  # ≈ 1.414
    """
    return last(within(tol, repeat1(f, x0)))
```

Edit: fix year mistake: the ancient civilization [Babylonia](https://en.wikipedia.org/wiki/Babylonia) flourished around 2000 BCE, not 4000 BCE (that would be the beginnings of [ancient Egypt](https://en.wikipedia.org/wiki/Ancient_Egypt)).

**Comment by Technologicat** [comment]
Args for `unpythonic.it.within` swapped in bfc8e21.

**Comment by Technologicat** [comment]
Resolved in 08b035a.
